### PR TITLE
Updates /resource/<catalog>/<kind>/<name> api for compatible version

### PR DESCRIPTION
**Pull request by sm43**
Updates preview and /v1 api

This adds a query param `minpipelinesversion` in api to search for
lates resource version compatible with pipelines version passed.

Signed-off-by: Shivam Mukhade <smukhade@redhat.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/main/CONTRIBUTING.md) for more details._

### CODE REVIEW

File: api/design/types/type.go [hunk 1]
```diff
@@ -18,6 +18,9 @@ import (
 	. "goa.design/goa/v3/dsl"
 )
 
+// Pipelines Version Regex
+var PipeliesVersionRegex = `^\d+(?:\.\d+){0,2}$`
```
Comment (piyush-garg) [inline api/design/types/type.go:22]: typo in name

File: api/design/resource.go [hunk 2]
```diff
@@ -180,13 +180,19 @@ var _ = Service("resource", func() {
 			Attribute("name", String, "Name of resource", func() {
 				Example("name", "buildah")
 			})
+			Attribute("minpipelinesversion", String, "To find resource compatible with a Tekton pipelines version, use this param", func() {
+				Pattern(types.PipeliesVersionRegex)
```
Comment (piyush-garg) [inline api/design/resource.go:184]: typo in name

File: api/v1/design/resource.go [hunk 3]
```diff
@@ -176,13 +176,19 @@ var _ = Service("resource", func() {
 			Attribute("name", String, "Name of resource", func() {
 				Example("name", "buildah")
 			})
+			Attribute("minpipelinesversion", String, "To find resource compatible with a Tekton pipelines version, use this param", func() {
+				Pattern(types.PipeliesVersionRegex)
```
Comment (piyush-garg) [inline api/v1/design/resource.go:180]: typo in name

**Review by PuneetPunamiya — COMMENTED** [review]
/lgtm
/meow

**Pull request merged**